File: Server/src/main/java/cn/arorms/hsi/server/exceptions/StorageException.java
package cn.arorms.hsi.server.exceptions;

/**
 * Refactored StorageException using Java.
 */
public class StorageException extends RuntimeException {
    
    public StorageException(String message) {
        super(message);
    }
    
    public StorageException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: Server/src/main/kotlin/cn/arorms/hsi/server/exceptions/StorageException.kt
package cn.arorms.hsi.server.exceptions

/**
 * Refactored StorageException using Kotlin.
 * * @param message The detail message.
 * @param cause The cause of the exception (optional, defaults to null).
 */
class StorageException @JvmOverloads constructor(
    message: String,
    cause: Throwable? = null
) : RuntimeException(message, cause)